File: app/models/base/VersionedModel.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.base;

import io.ebean.Model;
import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.Version;

@MappedSuperclass
public abstract class VersionedModel extends Model {

    @Version
    private long objectVersion;

    public long getObjectVersion() {
        return objectVersion;
    }

    public void setObjectVersion(long objectVersion) {
        this.objectVersion = objectVersion;
    }
}

File: app/models/base/VersionedModel.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.base

import io.ebean.Model
import jakarta.persistence.{MappedSuperclass, Version}

@MappedSuperclass
abstract class VersionedModel extends Model:
  @Version
  var objectVersion: Long = 0
